### DISCUSSION TITLE
How to use x:Static in StringFormat

### DISCUSSION THREAD

**Discussion by Supheria** [AvaloniaUI/Avalonia#17739] [Q&A] [answered]
Hello,
I want to format Binding text with localization like 

> Count: 10 // where Count is localization position

This is what I did:

```Xaml
...
xmlns:lang="clr-namespace:TestProject.Language"
...
Content = "{Binding Items.Length, StringFormat='{x:Static lang:Resource.ItemsCount}: {0}'}"
```

But it has an error on `: {0}` in StringFormat:

> Text after markup is not allowed

I also tried:

```Xaml
StringFormat='{}{x:Static lang:Resource.ItemsCount}: {0}'
```
But the content outputs nothing.
How to do this correctly?

**Comment by maxkatz6** [comment, marked as answer]
You can't combine extensions with strings inline like this.
You either need to create a resource with ": {0}" part pre-included.
- Or use MultiBinding, where string format is "{0}: {1}", and first binding defined similar to `<Binding Source="{x:Static lang:Resource.ItemsCount}" />`.
- Or create a custom format value converter, passing x:Static as a converter parameter.

**Reply by Supheria** [reply]
Thank you, it works. 

For reference:

```Xaml
<Label.Content>
   <MultiBinding StringFormat="{}{0}: {1}">
        <Binding Source="{x:Static lang:Resource.ItemsCount}" />
        <Binding Path="Items.Count" />
   </MultiBinding>
</Label.Content>
```
